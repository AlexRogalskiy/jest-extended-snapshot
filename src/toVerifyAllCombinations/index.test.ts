import matcher from "./";

expect.extend(matcher);

// This would be an existing function without tests.
// It has some twisted logic… but it works.
function myFunction(aNumber: number, aString: string): string {
  if (aNumber > 0) {
    return `${aString} #${aNumber}`;
  }

  if (aString === "foo") {
    return `${aNumber} bar`;
  }

  return "This is twisted…";
}

// Following test verifies all combinations of arguments
// to create a snapshot of function behaviour.
it("should continue working as before", () => {
  expect(myFunction).toVerifyAllCombinations([1, -1], ["random", "foo"]);
});
