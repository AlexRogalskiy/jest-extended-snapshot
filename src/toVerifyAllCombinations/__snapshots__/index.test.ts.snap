// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should continue working as before 1`] = `
Object {
  "-1,foo": "-1 bar",
  "-1,random": "This is twistedâ€¦",
  "1,foo": "foo #1",
  "1,random": "random #1",
}
`;
